server-timing/class-perflab-server-timing.php

```diff
@@ -195,18 +195,25 @@ static function( $value ) {
 	 * @return bool True if an output buffer should be used, false otherwise.
 	 */
 	public function use_output_buffer() {
-		/**
-		 * Filters whether an output buffer should be used to be able to gather additional Server-Timing metrics.
-		 *
-		 * Without an output buffer, it is only possible to cover metrics from before serving the template, i.e. before
-		 * the HTML output starts. Therefore sites that would like to gather metrics while serving the template should
-		 * enable this.
-		 *
-		 * @since 1.8.0
-		 *
-		 * @param bool $use_output_buffer Whether to use an output buffer.
-		 */
-		return apply_filters( 'perflab_server_timing_use_output_buffer', false );
+		if ( has_filter( 'perflab_server_timing_use_output_buffer' ) ) {
+			/**
+			 * Filters whether an output buffer should be used to be able to gather additional Server-Timing metrics.
+			 *
+			 * Without an output buffer, it is only possible to cover metrics from before serving the template, i.e. before
+			 * the HTML output starts. Therefore sites that would like to gather metrics while serving the template should
+			 * enable this.
+			 *
+			 * @since 1.8.0
+			 *
+			 * @param bool $use_output_buffer Whether to use an output buffer.
+			 */
+			return apply_filters( 'perflab_server_timing_use_output_buffer', false );
+		}
+
+		// Use the stored value from the Server-Timing screen.
+		$slug    = 'output_buffering';
+		$options = (array) get_option( PERFLAB_SERVER_TIMING_SETTING, array() );
+		return ! empty( $options[ $slug ] );
```

Comment:
Here I would advise to not call the filter conditionally but rather filter the value of the option at the end. That's a bit more typical in WP, rather than using a "short-circuit" filter.

Reply:
I was also wrestling with this. The reason why I went with the current approach is that _any_ filter present will cause the Output Buffering checkbox to be `disabled` on the settings screen. Therefore, it didn't seem to make sense to use the stored option. But it wouldn't really make much difference the other way around. Done in 8b415eb